src/environmentd/src/http.rs

```diff
@@ -529,36 +529,40 @@ pub struct AuthedClient {
 }
 
 impl AuthedClient {
-    async fn new(
-        adapter_client: &Client,
+    /// Note: the returned Future is intentionally boxed because it is very large.
```

Comment:
This is subjective, so it's ultimately up to you, but I think these comments are a little redundant. As long as the lint has a sufficient explanation somewhere around it, I think we can remove these comments.

Reply:
I agree! I assume people that wonder about this would try to remove the `boxed` call and find out the reason for it through the lint.